### PR TITLE
Update the packaged Bazel to 0.3.2, and also make it actually work.

### PR DESCRIPTION
**Pull request by IreneKnapp**
###### Motivation for this change

I completely rewrote the nix file for the Bazel build tool, which I don't think had ever resulted in an output that could be used to build things, although it at least compiled without error. It is now referring to the latest upstream.

I have manually tested that at least the cc_binary, cc_library, java_binary, and java_library rules work as they should; I don't have toolchains for things like Android or iOS installed here, so I leave it as future work to make sure everything else is usable.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/build-managers/bazel/java_stub_template.patch
Comment (benley) [inline pkgs/development/tools/build-managers/bazel/java_stub_template.patch]: You might want to add `set -o posix` if that could affect things in the stub script

**Review by benley — APPROVED** [review]
This looks great - I'll wait for another reviewer to look over the changes, but FHSUserEnv does seem like the right approach to me.

**Comment by joachifm** [comment]
cc maintainer @philandstuff

File: pkgs/development/tools/build-managers/bazel/default.nix
Comment (Mic92) [inline pkgs/development/tools/build-managers/bazel/default.nix]: i686 also worked for me. I think it is fair to say:

```
platforms = platforms.linux;
```

**Comment by teh** [comment]
lgtm from me, too

**Comment by philandstuff** [comment]
I'm happy with this.

**Comment by joachifm** [comment]
Still failing on hydra, e.g. https://hydra.nixos.org/build/42550256/log/raw

**Comment by IreneKnapp** [comment]
Huh. From the log, it's not obvious to me why it would be failing. I hope to have time to debug further sometime this week; any pointers on how Hydra differs from a local environment would be appreciated.

**Comment by joachifm** [comment]
@IreneKnapp you've indicated having built the package without sandboxing. If so, that's a significant source of variability.

**Comment by joachifm** [comment]
It looks like https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/build-fhs-userenv/chroot-user.rb#L43 is where the error originates. @abbradar may know why.

**Comment by abbradar** [comment]
Wow, I haven't expected that someone would use it like this! Certainly interesting.

I'll try to debug this later -- do I understand correctly that this error is not visible when built on a local machine with a sandbox enabled?

**Comment by IreneKnapp** [comment]
Huh. The cause isn't obvious to me from that log. I can take a look
sometime this week.

**Pull request merged**
